src/libs/Navigation/NavigationRoot.js

```diff
@@ -0,0 +1,135 @@
+import _ from 'underscore';
+import React, {Component} from 'react';
+import {Linking} from 'react-native';
+import PropTypes from 'prop-types';
+import {
+    getStateFromPath,
+    getPathFromState,
+    NavigationContainer,
+} from '@react-navigation/native';
+import Onyx from 'react-native-onyx';
+import {navigationRef} from './Navigation';
+import linkingConfig from './linkingConfig';
+import AppNavigator from './AppNavigator';
+import modalRoutesPropTypes from './AppNavigator/modalRoutesPropTypes';
+import mainRoutesPropTypes from './AppNavigator/mainRoutesPropTypes';
+import sidebarRoutePropTypes from './AppNavigator/sidebarRoutePropTypes';
+import publicRoutePropTypes from './AppNavigator/publicRoutePropTypes';
+import getPathName from './getPathName';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+
+const propTypes = {
+    modalRoutes: modalRoutesPropTypes.isRequired,
+    mainRoutes: mainRoutesPropTypes.isRequired,
+    sidebarRoute: sidebarRoutePropTypes.isRequired,
+    publicRoutes: PropTypes.arrayOf(publicRoutePropTypes).isRequired,
+    authenticated: PropTypes.bool.isRequired,
+    currentlyViewedReportID: PropTypes.string,
+};
+
+const defaultProps = {
+    currentlyViewedReportID: null,
+};
+
+class NavigationRoot extends Component {
+    constructor(props) {
+        super(props);
+
+        this.initialState = undefined;
+        this.state = {
+            loading: true,
+            initialState: undefined,
+            drawerOpenByDefault: true,
+        };
+    }
+
+    componentDidMount() {
+        Linking.getInitialURL()
```

Comment:
sorry yes this would only apply to mobile, if a user clicks a deeplink you need the listener set up to catch the url being passed into the app and route them (if the app is already running)